toke.c

```diff
@@ -4685,6 +4718,15 @@ S_intuit_more(pTHX_ char *s, char *e)
             break;
         }   /* End of switch */
 
+        /* khw: 'seen' is declared as a char.  This ++ can cause it to wrap.
+         * This gives different results with compilers for which a plain 'char'
+         * is actually unsigned, versus those where it is signed.  I believe it
+         * is undefined behavior to wrap a 'signed'.  I think it should be
+         * instead declared an unsigned int to make the chances of wrapping
+         * essentially zero.
```

Comment:
I think making it unsigned would be good. Having it wrap at 8 bits just means the 256'th occurrence adds 0 to the weight, which is no great loss - making them uints doesn't make any difference to the heuristic in any realistic scenario.